### DISCUSSION TITLE
Input inside a Panel(rich)

### DISCUSSION THREAD

**Discussion by RougherO** [Textualize/rich#1980] [Q&A] [answered]
Is it possible to accept input inside a panel 

For a bit of context I am trying to make a command line contact book for my school project and I want to provide an area for the user to type in contact details or up down arrow keys to move through the options however I'm facing a difficulty to replicate this.

I also thought of dividing the screen into two layouts top for display and bottom for viewing what I'm typing but I can't figure out how to do that..

I am trying to do this while in a Live context manager
(with Live(...) as live:) 

Thanks in advance 😃

EDIT : I am thinking of integrating curses with rich for the input .... divide the screen into 2 windows and pass these windows to the rich renderer to render the output onto

**Comment by willmcgugan** [comment, marked as answer]
Rich has no support for getting input at the moment. But in theory if you add input you can implement what you describe.

**Reply by RougherO** [reply]
Oh ok
No problem, I looked into textual as it's very promising so I think I'll wait for it to be finished until then I am looking into other CUI/TUI libraries
